### ISSUE TITLE
404 Not Found

### ISSUE THREAD

**Issue by NurdTurd** [yudai/gotty#39]
`gotty --random-url --permit-write top --addr 127.0.0.1`

When I go to the page. I get 404 not found.

**Comment by yudai** [comment]
Did you add :8080?

**Comment by NurdTurd** [comment]
Yes.

**Comment by yudai** [comment]
Could you provide the URL you accessed?

**Comment by NurdTurd** [comment]
Sure.

http://127.0.0.1:8080/6kbv8603

I was using `--random-url`

**Comment by yudai** [comment]
Cloud you try the latest pre-release? It has some changes in handling URLs that might solves your issue.

**Comment by NurdTurd** [comment]
Sure. :)

**Comment by NurdTurd** [comment]
@yudai How would I go about this? I'm on a Mac.

**Comment by yudai** [comment]
Try this one: https://github.com/yudai/gotty/releases/download/pre-release/darwin_amd64.tar.gz

**Comment by NurdTurd** [comment]
Thank you.

**Comment by NurdTurd** [comment]
Using the one you sent me: https://shel.io/f/aiUg.png

**Comment by yudai** [comment]
Could you try `gotty --random-url --permit-write --addr 127.0.0.1 --port 8080 top`?

It looks like there is another bug in parsing arguments

**Comment by NurdTurd** [comment]
Error(s): https://shel.io/f/5ppd.png

**Comment by yudai** [comment]
Thanks for testing. I'll test on Mac and try to figure out what's happening soon. Give me some time.

**Comment by NurdTurd** [comment]
No problem. I am here to test anything you need. :)

**Comment by yudai** [comment]
I could not reproduce your problem.
However, I added an error handling to the code around you got the error. Could you redownload the latest one from the same URL and try it? And I'd like to see what happens when you run gotty without sudo.

**Comment by nzyuzin** [comment]
I had just the same problem before I updated to current revision (from 94f571d to 56ae9ab).
It seems something in this range has fixed the issue.

@NurdTurd you could try fetching last changes to see if it's fixed for you too.

**Comment by NurdTurd** [comment]
Sure. I got this: https://shel.io/f/8BsP.png

**Comment by nzyuzin** [comment]
Same for me for the release binary.
Works fine with binary built with `go get github.com/yudai/gotty` though.

**Comment by yudai** [comment]
Found the root cause: http://stackoverflow.com/questions/20609415/cross-compiling-user-current-not-implemented-on-linux-amd64

**Comment by yudai** [comment]
Pushed the fix to the latest release. I hope it will work on your environment.

**Comment by NurdTurd** [comment]
Thanks!

**Comment by NurdTurd** [comment]
Using the new one, another 404 error. https://shel.io/f/MjMH.png

**Comment by yudai** [comment]
Hi, thank you for the report. Unfortunately, I cannot reproduce the issue you are experience.
Could you paste the full log? If your browser reaches the gotty server, gotty should output some GET lines like below:

```
./gotty -a 127.0.0.1 -r top
2015/08/24 12:22:15 Server is starting with command: top
2015/08/24 12:22:15 URL: http://127.0.0.1:8080/17qe10bp/
2015/08/24 12:22:33 GET /17qe10bp/
2015/08/24 12:22:33 GET /17qe10bp/hterm.js
2015/08/24 12:22:34 GET /17qe10bp/gotty.js
2015/08/24 12:22:34 GET /17qe10bp/ws
2015/08/24 12:22:34 New client connected: 127.0.0.1:55083
2015/08/24 12:22:34 Command is running for client 127.0.0.1:55083 with PID 45432
2015/08/24 12:22:47 GET /invalid_token/
2015/08/24 12:22:47 Command exited for: 127.0.0.1:55083
2015/08/24 12:22:47 Connection closed: 127.0.0.1:55083
```

In this case, the 404 error page was shown by accessing `http://127.0.0.1:8080/invalid_token/` and that was logged as `2015/08/24 12:22:47 GET /invalid_token/` .

**Comment by yudai** [comment]
Closing as no activities last week. Please reopen if this issue still exists.
